src/main/java/seedu/address/model/recipe/Ingredient.java

```diff
@@ -0,0 +1,109 @@
+package seedu.address.model.recipe;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.util.Objects;
+
+/**
+ * Represents a Recipe's ingredient in the recipe book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidIngredientName(String)},
+ * {@link #isValidQuantity(String)}, and {@link #isValidQuantifier(String)}.
+ */
+public class Ingredient {
+
+    public static final String NAME_CONSTRAINTS = "Ingredients name should not be left blank";
+    public static final String QUANTITY_CONSTRAINTS = "Quantity should be a valid integer and should not be left blank";
+    public static final String QUANTIFIER_CONSTRAINTS = "Quantifier name should not be left blank";
+
+    /*
+     * The first character of the ingredient must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String NAME_VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+
+    /*
+     * The quantity of the ingredient must be a valid number.
+     */
+    public static final String QUANTITY_VALIDATION_REGEX = "\\d";
+
+    /**
+     * The first character of the quantifier must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String QUANTIFIER_VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+
+    public final String ingredientName;
+    public final double quantity;
+    public final String quantifier;
+
+    /**
+     * Constructs a {@code Ingredient}.
+     *
+     * @param name A valid name.
+     * @param quantity A valid quantity.
+     * @param quantifier A valid quantifier.
+     */
+    public Ingredient(String name, String quantity, String quantifier) {
```

Comment:
I think quantity should be of type double here instead of String here.

Reply:
No wait, I didn't see that you are parsing the String to Integer in line 59. But what about quantities specified by user that cannot be parsed as Integer? e.g 0.5 litres of milk

Reply:
I think it's a better idea to parse as Double then. 

I'll probably need some help/feedback on this when we integrate with Parser. Tut 2 parsing workflow was not as simple as I had originally thought this portion would be, and I think it's better to try and resolve this issue when we get to that part so we don't have to make _too_ many iterations.